src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -639,14 +639,21 @@ struct
       (* Whether [con1] contains a var in [env]. *)
       let env_exists_mem_con1 env con1 =
         try
-          Lincons1.iter (fun _ var ->
-              if Environment.mem_var env var then
+          Lincons1.iter (fun coeff var ->
+              (* Lincons1 from polyhedra may contain variable with zero coefficient.
+                 These are silently not printed! *)
+              if not (Coeff.is_zero coeff) && Environment.mem_var env var then
                 raise Not_found
             ) con1;
           false
         with Not_found ->
           true
```

Comment:
I was confused by `Not_found` here. Should we use a custom exception here such as `Found` to make it easier to understand what's happening?

Reply:
You're right, it slightly confused me too. I'll use `Stdlib.Exit` here as we already do in other places.